pkg/services/ngalert/state/historian/annotation.go

```diff
@@ -45,6 +46,9 @@ func (h *AnnotationStateHistorian) RecordStates(ctx context.Context, rule *ngmod
 func (h *AnnotationStateHistorian) buildAnnotations(rule *ngmodels.AlertRule, states []state.StateTransition, logger log.Logger) []annotations.Item {
 	items := make([]annotations.Item, 0, len(states))
 	for _, state := range states {
+		if !needsLogging(state) {
```

Comment:
Perhaps something like this? I don't think we need to disable debug logging for non transitioning states?

```suggestion
		if !shouldAnnotate(state) {
```

Reply:
Renamed!